core/common/src/main/java/alluxio/util/CommonUtils.java

```diff
@@ -227,6 +227,28 @@ public static void waitFor(String description, Function<Void, Boolean> condition
   }
 
   /**
+   * Waits for an operation to return a non-null value with a specified timeout.
+   *
+   * @param description the description of this operation
+   * @param operation the operation
+   * @param timeoutMs the timeout
+   * @param <T> the type of the return value
+   * @return the return value, null if it times out
+   */
+  public static <T> T waitFor(String description, Function<Void, T> operation, long timeoutMs) {
```

Comment:
instead of adding all these different versions of `waitFor`, there should only be two ... one that only take required arguments and one with required argument plus `WaitForOptions` argument